src/main/java/appeng/integration/modules/CraftGuide.java

```diff
@@ -152,56 +158,51 @@ else if( recipe instanceof ShapelessRecipe )
 		}
 	}
 
-	private void addGrinderRecipes( RecipeTemplate template, RecipeGenerator generator )
-	{
-
-	}
-
-	private void addInscriberRecipes( RecipeTemplate template, RecipeGenerator generator )
-	{
-
-	}
-
-	@Override
-	public RecipeTemplate createRecipeTemplate( Slot[] slots, ItemStack craftingType )
-	{
-		return this.parent.createRecipeTemplate( slots, craftingType );
-	}
-
-	@Override
-	public RecipeTemplate createRecipeTemplate( Slot[] slots, ItemStack craftingType, String backgroundTexture, int backgroundX, int backgroundY, int backgroundSelectedX, int backgroundSelectedY )
-	{
-		return this.parent.createRecipeTemplate( slots, craftingType, backgroundTexture, backgroundX, backgroundY, backgroundSelectedX, backgroundSelectedY );
-	}
-
-	@Override
-	public RecipeTemplate createRecipeTemplate( Slot[] slots, ItemStack craftingType, String bgTexture, int bgX, int bgY, String selectedBGTexture, int selectedBGX, int selectedBGY )
-	{
-		return this.parent.createRecipeTemplate( slots, craftingType, bgTexture, bgX, bgY, selectedBGTexture, selectedBGX, selectedBGY );
-	}
-
-	@Override
-	public void addRecipe( RecipeTemplate template, Object[] crafting )
+	private void addGrinderRecipes( IAppEngApi api, ItemStack grindstone, RecipeGenerator generator )
 	{
-		this.parent.addRecipe( template, crafting );
+		ItemStack handle = api.definitions().blocks().crankHandle().maybeStack(1).orNull();
+		Slot[] grinderSlots = new ItemSlot[] {
+				new ItemSlot( 3, 21, 16, 16 ).drawOwnBackground(),
+				new ItemSlot( 41, 21, 16, 16, true ).drawOwnBackground().setSlotType( SlotType.OUTPUT_SLOT ),
+				new ChanceSlot( 59, 12, 16, 16, true).setRatio( 10000 ).setFormatString( " (%1$.2f%% chance)" ).drawOwnBackground().setSlotType( SlotType.OUTPUT_SLOT ),
+				new ChanceSlot( 59, 30, 16, 16, true).setRatio( 10000 ).setFormatString( " (%1$.2f%% chance)" ).drawOwnBackground().setSlotType( SlotType.OUTPUT_SLOT ),
+				new ExtraSlot( 22, 12, 16, 16, handle ).clickable( true ).showName( true ).setSlotType( SlotType.MACHINE_SLOT ),
+				new ExtraSlot( 22, 30, 16, 16, grindstone ).clickable( true ).showName( true ).setSlotType( SlotType.MACHINE_SLOT ) };
```

Comment:
Could the be extracted to a constant similar to `smallCraftingSlots`? Or can it not share the same instance, which makes `smallCraftingSlots` also questionable? It is a slight performance and memory regression when creating an actual constant object for each recipe.